packages/core/lib/FileSystemCache.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+const {join} = require('path');
+const {rmdir, readdir, readFile, writeFile, unlink, existsSync, mkdirSync} = require('fs');
+const {promisify} = require('util');
+const crypto = require('crypto');
+
+const path = require('path');
+const readFileAsync = promisify(readFile);
+const writeFileAsync = promisify(writeFile);
+const readdirAsync = promisify(readdir);
+const rmdirAsync = promisify(rmdir);
+const unlinkAsync = promisify(unlink);
+
+class FileSystemCache {
+  static get(
+    opts = {
+      baseDir: join(__dirname, '.cache'),
+    }
+  ) {
+    return new FileSystemCache(opts);
+  }
+
+  constructor(opts) {
+    this.opts = opts;
+    this.cache = new Map();
+  }
+
+  async get(key, defaultValue = null) {
+    let value = this.cache.get(key);
+    if (value) {
+      return value;
+    }
+    const cacheFile = this.createCacheFileName(key);
+    try {
+      const content = await readFileAsync(cacheFile, 'utf-8');
+      value = JSON.parse(content);
+      this.cache.set(key, value);
+    } catch (error) {
+      value = defaultValue;
+    }
+    return value;
+  }
+
+  async set(key, value) {
+    const cacheFile = this.createCacheFileName(key);
+    try {
+      this.cache[key] = value;
```

Comment:
Hm. That's been tested already

Reply:
But changed to `set`

Reply:
I think the test worked because it was always getting a cache miss and loading from a file. You could maybe add a test that clears the folder on the filesystem after a `set` to make sure it loads it from the LRU.